### PR TITLE
VMware: Update vmware_guest.py  Add option to modify disk type while cloning template

### PR DESCRIPTION
**Pull request by chris93111**
##### SUMMARY
Add parameters to module vmware_guest for conversion of disk to thin or thick when vm is cloned or deployed with template 

@Akasurde 

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
vmware_guest.py

##### ANSIBLE VERSION

```paste below
2.8-devel
```

### CODE REVIEW

**Comment by Akasurde** [comment]
@chris93111 Thanks for the PR. Could you please address the https://github.com/ansible/ansible/pull/47859#issuecomment-434688013 issues ? I will test this tomorrow and let you know the feedback.

File: lib/ansible/modules/cloud/vmware/vmware_guest.py [hunk 1]
```diff
@@ -2305,6 +2324,7 @@ def main():
         customization_spec=dict(type='str', default=None),
         vapp_properties=dict(type='list', default=[]),
         datastore=dict(type='str'),
+        convert=dict(type='str'),
```
Comment (Akasurde) [inline lib/ansible/modules/cloud/vmware/vmware_guest.py]: Add documentation for this parameter with `version_added`

Comment (chris93111) [reply lib/ansible/modules/cloud/vmware/vmware_guest.py]: ok is commit

File: lib/ansible/modules/cloud/vmware/vmware_guest.py [hunk 2]
```diff
@@ -2305,6 +2332,7 @@ def main():
...
```
Comment (Akasurde) [inline lib/ansible/modules/cloud/vmware/vmware_guest.py]:
```suggestion
        convert=dict(type='str', choices=['thin', 'thick', 'eagerzeroedthick']),
```

File: lib/ansible/modules/cloud/vmware/vmware_guest.py [hunk 3]
```diff
@@ -2035,6 +2043,25 @@ def deploy_vm(self):
                     relospec.host = self.select_host()
                 relospec.datastore = datastore
 
+                # Convert disk present in template if is set
+                if self.params['convert']:
+                    if self.params['convert'] not in ['thin', 'thick', 'eagerzeroedthick']:
```
Comment (Akasurde) [inline lib/ansible/modules/cloud/vmware/vmware_guest.py]: With `choices` you don't need this check.

**Review by Akasurde — CHANGES_REQUESTED** [review]
Could you please add unit test for this change ?

**Comment by Akasurde** [comment]
@pdellaert @dericcrago @Im0 @tchernomax @ckotte Could you please review this ?

File: lib/ansible/modules/cloud/vmware/vmware_guest.py (hunk 3 above)
Comment (chris93111) [reply lib/ansible/modules/cloud/vmware/vmware_guest.py]: Ok i change this thanks

**Comment by chris93111** [comment]
@Akasurde  can you please tell me if unit test it's valid please ?

File: lib/ansible/modules/cloud/vmware/vmware_guest.py (hunk 2 above)
Comment (chris93111) [reply lib/ansible/modules/cloud/vmware/vmware_guest.py]: ok

**Review by chris93111 — COMMENTED** [review]
correction documentation choices and remove check

**Comment by tgoetheyn** [comment]
I tried testing but I get
`TypeError: find_obj() got an unexpected keyword argument 'folder'`

I am probably doing something wrong because it is the first time I try to test a module.
I downloaded the module in a testfolder and pointed my "library" to that folder in the ansible.cfg file.
Am I forgetting something?

**Comment by chris93111** [comment]
Hi @tgoetheyn what version ansible you have ? If you would like test the module you need to uptate also the vmware.py

**Comment by chris93111** [comment]
Hi @tgoetheyn what version ansible you have ? If you would like test the module you need to uptate also the vmware.py

**Comment by tgoetheyn** [comment]
Hi @chris93111 
I updated the vmware.py and now it works.

I tried to deploy a VM from a template that is located on an NFS volume and now the disk type is successfully converted to "eagerzeroedthick"

So for me it looks ok.

Thanks for the work you put into this!

**Comment by chris93111** [comment]
Thanks for you return @tgoetheyn

**Comment by chris93111** [comment]
@Akasurde new unit test for this change

**Review by chris93111 — COMMENTED** [review]
Unit test for clone with clone

**Review by chris93111 — COMMENTED** [review]
Ok

**Review by Akasurde — APPROVED** [review]

**Comment by Akasurde** [comment]
@pdellaert @dericcrago @Im0 @tchernomax @ckotte Could you please review this ?

**Comment by Akasurde** [comment]
rebuild_merge

**Comment by Akasurde** [comment]
@chris93111 Thanks for the contribution.

**Pull request merged**